File: gmlrva-lib/src/main/java/pt/simdea/gmlrva/lib/diff/GenericPayload.java
/*
 * Copyright (c) 2017. Simdea.
 */

package pt.simdea.gmlrva.lib.diff;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Magic Constant Annotation Enum containing the possible {@link GenericPayload} change key options.
 *
 * Created by Paulo Ribeiro on 9/26/2017.
 * Simdea © All Rights Reserved.
 * paulo.ribeiro@simdea.pt
 */
@Retention(RetentionPolicy.SOURCE)
@StringDef({
        GenericPayload.UPDATE_ITEM
})
public @interface GenericPayload {

    /* Constants */

    /** UPDATE_ITEM representing an item that needs to be updated. */
    String UPDATE_ITEM = "UPDATE_ITEM";

}

File: gmlrva-lib/src/main/java/pt/simdea/gmlrva/lib/diff/GenericPayload.kt
/*
 * Copyright (c) 2017. Simdea.
 */

package pt.simdea.gmlrva.lib.diff

import androidx.annotation.StringDef
import pt.simdea.gmlrva.lib.diff.Const.Companion.UPDATE_ITEM

/**
 * Magic Constant Annotation Enum containing the possible [GenericPayload] change key options.
 *
 * Created by Paulo Ribeiro on 9/26/2017.
 * Simdea © All Rights Reserved.
 * paulo.ribeiro@simdea.pt
 */
@Retention(AnnotationRetention.SOURCE)
@StringDef(UPDATE_ITEM)
annotation class GenericPayload

class Const {
    companion object {
        /* Constants */

        /** UPDATE_ITEM representing an item that needs to be updated.  */
        const val UPDATE_ITEM = "UPDATE_ITEM"
    }
}

